### PR TITLE
DCMAW-8966: generate STS mock key pair and publish keys to S3

### PR DESCRIPTION
**Pull request by anahickmann**
​DCMAW-8966
### What changed
- Add a script that:
  - Generates an asymmetric key pair for signing
  - Saves the private key to the `jwks` S3 bucket in JWK format
  - Saves the public key to the  `jwks` S3 bucket `/.well-known` directory in JWKS format 

**Tested in the `dev` environment**
https://github.com/user-attachments/assets/c3f6d1f3-4acc-4e98-97a5-e5d4be8ca485

**Public key**
Screenshot 2024-08-19 at 16 46 52

**Private key**
Screenshot 2024-08-19 at 16 52 36

### Why did it change
- Private key is required for signing the token issued by the STS mock
- JWKS (public key) is required for validating the signature on the token issued by the STS mock

## Checklists

- [x] There is a ticket raised for this PR that is present in the branch name
- [x] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [ ] Demo to a BA, TA, and the team.
- [x] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

**Review by J-son1 — APPROVED** [review]

**Comment by sandymay-dd** [comment]
> **Question:** I'm not sure if we want the new directory `jwks-helper-script` to live inside the `sts-mock` directory or outside 🤔

inside for now - until we know we might want to use it for more than one application

**Comment by sandymay-dd** [comment]
Please may you delete this public/private keypair if you haven't just to make sure you don't use it for anything :)

**Comment by sandymay-dd** [comment]
> > Please may you delete this public/private keypair if you haven't just to make sure you don't use it for anything :)
> 
> Which key pair?

the one in the evidence :)

**Comment by sandymay-dd** [comment]
> > > > Please may you delete this public/private keypair if you haven't just to make sure you don't use it for anything :)
> > > 
> > > 
> > > Which key pair?
> > 
> > 
> > the one in the evidence :)
> 
> Ohhhh! 🤦 - I will delete it from the bucket

thank you! Public repo :)

File: sts-mock/jwks-helper-script/publish_keys_to_s3.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+fi
+
+echo "Generating keys"
+npm install
+node jwksHelperScript.mjs $PRIVATE_KEY_FILE_NAME $JWKS_FILE_NAME
+
+STACK_NAME="$1"
+
+echo "Uploading keys"
+aws s3 cp $PRIVATE_KEY_FILE_NAME s3://"$STACK_NAME"-jwks/
```
Comment (sandymay-dd) [inline sts-mock/jwks-helper-script/publish_keys_to_s3.sh:24]: Do we need to create an S3 bucket before doing this?

Comment (sandymay-dd) [reply sts-mock/jwks-helper-script/publish_keys_to_s3.sh:24]: let me run away now - I completely forgot we already had a template!

**Review by sandymay-dd — APPROVED** [review]

**Pull request merged**